src/main/java/fi/aalto/cs/apluscourses/intellij/actions/SubmitExerciseAction.java

```diff
@@ -161,6 +163,11 @@ public void update(@NotNull AnActionEvent e) {
       e.getPresentation().setEnabled(project != null
           && authentication != null && courseViewModel != null
           && (isSubmittableExerciseSelected || isSubmittableSubmissionSelected));
+      if (progressViewModel.getCurrentProgress() != null && !progressViewModel.getCurrentProgress().isFinished()) {
```

Comment:
I think a better check is something like `selectedItem.getModel() instanceOf DummyExercise` or DummySubmissionResult

Reply:
Because all the assignments don't have to be loaded to submit assignments